### PR TITLE
Add an option to log meta data at the top level

### LINKED ISSUE #34: Meta data should be logged at the top level
Given this code:

    logger.log({
        level: 'info',
        message: 'My message',
        additional: 'More stuff'
    });

Other transports, such as File transport, log this JSON:

    {"level":"info","message":"My message","additional":"More stuff"}

Whereas winston-redis logs this JSON:

    {"level":"info","message":"My message","meta":{"additional":"More stuff"}}

Is it possible to add an option to log the meta data at the top level rather than as a child property?

### PR DESCRIPTION
**Pull request by mooski**
Closes #34

### CODE REVIEW

File: lib/winston-redis.js [hunk 1]
```diff
@@ -39,12 +39,13 @@ class Redis extends Transport {
...
-    this.channel   = options.pchannel || options.channel;
-    this.pattern   = options.pattern || !!options.pchannel;
-    this.metadata  = options.meta || {};
+    this.json              = options.json !== false;
+    this.length            = options.length || 200;
+    this.container         = options.container || 'winston';
+    this.channel           = options.pchannel || options.channel;
+    this.pattern           = options.pattern || !!options.pchannel;
+    this.metadata          = options.meta || {};
+    this.logMetaAtTopLevel = options.logMetaAtTopLevel || false;
```
Comment (mgibeau) [inline lib/winston-redis.js]: What would you think of `flatMeta` instead?

**Comment by mark-storefeeder** [comment]
I've changed the property name as you suggested and added a description of the property to the readme.

One thing I found peculiar is that the tests are behaving differently to my own test implementation. Given that I've set `flatMeta` to true, the tests don't log any meta data, whereas my own implementation logs this data as expected. I'm not sure why.

There's an obvious alternative implementation that would solve this problem (`JSON.stringify({ level, message, ...meta })`), but it seems like extra unnecessary work just for the benefit of the tests.

**Comment by mooski** [comment]
I've changed the property name as you suggested and added a description of the property to the readme.

One thing I found peculiar is that the tests are behaving differently to my own test implementation. Given that I've set `flatMeta` to true, the tests don't log any meta data, whereas my own implementation logs this data as expected. I'm not sure why.

There's an obvious alternative implementation that would solve this problem (`JSON.stringify({ level, message, ...meta })`), but it seems like extra unnecessary work just for the benefit of the tests.

**Comment by mgibeau** [comment]
I'm good the the changes as is, will take a look at the tests, I've been meaning to improve them for the longest time anyways :) Thanks!

**Comment by mooski** [comment]
Thanks for merging this!

Would it be possible to push the latest version to NPM so we can consume this change?

**Comment by nfedyk** [comment]
Thanks for doing this. Is there are plan for a release version so this feature can be utilized?

**Comment by mark-storefeeder** [comment]
@nfedyk It's not ideal, but we needed this feature too so I took this liberty of publishing it to npm here if that helps: https://www.npmjs.com/package/@mooski/winston-redis

**Comment by mooski** [comment]
@nfedyk It's not ideal, but we needed this feature too so I took this liberty of publishing it to npm here if that helps: https://www.npmjs.com/package/@mooski/winston-redis

**Comment by mgibeau** [comment]
Sorry this took a while, I just published the latest version to npm: https://www.npmjs.com/package/winston-redis/v/3.1.0

**Pull request merged**